presto-hive/src/main/java/com/facebook/presto/hive/util/InternalHiveSplitFactory.java

```diff
@@ -186,16 +179,15 @@ public InternalHiveSplitFactory(
                 readBucketNumber,
                 tableBucketNumber,
                 splittable,
-                forceLocalScheduling && allBlocksHaveRealAddress(blocks),
                 (nodeSelectionStrategy == HARD_AFFINITY && !allBlocksHaveRealAddress(blocks)) ? NO_PREFERENCE : nodeSelectionStrategy,
```

Comment:
I think this should be 
```java
forceLocalScheduling && allBlocksHaveRealAddress(blocks) && nodeSelectionStrategy == HARD_AFFINITY ? HARD_AFFINITY : nodeSelectionStrategy
```
Because `forceLocalScheduling` can be turned off